Magento/ruleset.xml

```diff
@@ -5,202 +5,438 @@
-        <severity>6</severity>
-    </rule>
-    <rule ref="Magento.NamingConvention.ReservedWords">
-        <severity>10</severity>
+        <severity>8</severity>
+        <type>warning</type>
     </rule>
     <rule ref="Magento.PHP.DateTime">
-        <severity>6</severity>
-    </rule>
-    <rule ref="Magento.PHP.DiscouragedFunction">
         <severity>8</severity>
+        <type>warning</type>
     </rule>
-    <rule ref="Magento.PHP.FinalImplementation">
-        <severity>10</severity>
+    <rule ref="Magento.PHP.ShortEchoSyntax">
+        <severity>8</severity>
+        <type>warning</type>
     </rule>
-    <rule ref="Magento.PHP.Goto">
-        <severity>10</severity>
+    <rule ref="Magento.Templates.ThisInTemplate">
+        <include-pattern>*.phtml</include-pattern>
+        <severity>8</severity>
+        <type>warning</type>
     </rule>
-    <rule ref="Magento.PHP.LiteralNamespaces">
+    <rule ref="Magento.Translation.ConstantUsage">
         <severity>8</severity>
+        <type>warning</type>
     </rule>
-    <rule ref="Magento.PHP.ReturnValueCheck">
-        <severity>10</severity>
+
+    <!-- Severity 7 warnings: General code issues. -->
+    <rule ref="Generic.Arrays.DisallowLongArraySyntax">
+        <severity>7</severity>
+        <type>warning</type>
     </rule>
-    <rule ref="Magento.PHP.ShortEchoSyntax">
-        <severity>8</severity>
+    <rule ref="Generic.Classes.DuplicateClassName">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.CodeAnalysis.ForLoopShouldBeWhileLoop">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.CodeAnalysis.ForLoopWithTestFunctionCall">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.CodeAnalysis.JumbledIncrementer">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.CodeAnalysis.UnconditionalIfStatement">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.CodeAnalysis.UselessOverridingMethod">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.Metrics.CyclomaticComplexity">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.Metrics.NestingLevel">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.PHP.DisallowShortOpenTag">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Generic.PHP.DisallowShortOpenTag.EchoFound">
+        <severity>0</severity>
+    </rule>
+    <rule ref="Magento.CodeAnalysis.EmptyBlock">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Magento.Exceptions.Namespace">
+        <severity>7</severity>
+        <type>warning</type>
+    </rule>
+    <rule ref="Magento.PHP.DiscouragedFunction">
```

Comment:
Per our discussion, let's increase severity to 9 because it checks for functions that may lead to security issues.